### PR TITLE
added a usage example to average pooling 2d layer

### PR DESCRIPTION
**Pull request by generationzcode**
I added a usage example to average pooling layer 2d.

### CODE REVIEW

**Review by rpalakkal — CHANGES_REQUESTED** [review]
Nice job on this usage example! 
Make sure to add space between the DocTest and the other sections so that it is testable. For reference, here is DocTest Format guide: https://www.tensorflow.org/community/contribute/docs_ref. Also, you can look at the usage example I added #34829.

File: tensorflow/python/keras/layers/pooling.py [hunk 1]
```diff
@@ -479,6 +479,9 @@ class AveragePooling2D(Pooling2D):
       4D tensor with shape `(batch_size, pooled_rows, pooled_cols, channels)`.
     - If `data_format='channels_first'`:
       4D tensor with shape `(batch_size, channels, pooled_rows, pooled_cols)`.
+  Usage Example:
```
Comment (rpalakkal) [inline tensorflow/python/keras/layers/pooling.py]: add spacing

Comment (WilliamHYZhang) [inline tensorflow/python/keras/layers/pooling.py]: "Usage Example" seems redundant

File: tensorflow/python/keras/layers/pooling.py [hunk 2]
```diff
@@ -479,6 +479,9 @@ class AveragePooling2D(Pooling2D):
...
+    python
```
Comment (WilliamHYZhang) [inline tensorflow/python/keras/layers/pooling.py]: This line is not needed

File: tensorflow/python/keras/layers/pooling.py [hunk 3]
```diff
@@ -479,6 +479,9 @@ class AveragePooling2D(Pooling2D):
...
+    >>>Average_pooling_layer = tf.keras.layers.AveragePooling2D(pool_size = (2,2),strides = (1,1),padding = "same")
```
Comment (WilliamHYZhang) [inline tensorflow/python/keras/layers/pooling.py]: Please follow conventional variable naming, ie `average_pooling_layer`

Comment (WilliamHYZhang) [reply tensorflow/python/keras/layers/pooling.py]: Also, please add what `average_pooling_layer` returns in order to make this a doctest

**Review by WilliamHYZhang — COMMENTED** [review]
Nice work, added some comments in addition to @rpalakkal's.

**Comment by generationXcode** [comment]
Done!

**Comment by generationXcode** [comment]
Thanks!

File: tensorflow/python/keras/layers/pooling.py [hunk 4]
```diff
@@ -479,6 +479,9 @@ class AveragePooling2D(Pooling2D):
       4D tensor with shape `(batch_size, pooled_rows, pooled_cols, channels)`.
     - If `data_format='channels_first'`:
       4D tensor with shape `(batch_size, channels, pooled_rows, pooled_cols)`.
+  
+  >>>average_pooling_layer = tf.keras.layers.AveragePooling2D(pool_size = (2,2),strides = (1,1),padding = "same")
```
Comment (WilliamHYZhang) [inline tensorflow/python/keras/layers/pooling.py]: Please see [here](https://www.programcreek.com/python/example/89711/keras.layers.AveragePooling2D) for an example of how to use `tf.keras.layers.AveragePooling2D`. You need to call it on a tensor, ie

```
x = tf.keras.Input(shape=(3, 3, 1), name='x')
y = average_pooling_layer = tf.keras.layers.AveragePooling2D(pool_size = (2,2),strides = (1,1),padding = "same")(x)
```

In this example, `y` would return
```
Tensor("average_pooling2d_9/Identity:0", shape=(None, 3, 3, 1), dtype=float32)
```
I hope this makes sense :)

**Comment by WilliamHYZhang** [comment]
Great job so far, please do make sure to use the function correctly, as of now your `average_pooling_layer` is just the `tf.keras.layers.AveragePooling2D` object.

File: tensorflow/python/keras/layers/pooling.py (hunk 4 above)
Comment (generationXcode) [reply tensorflow/python/keras/layers/pooling.py]: Ohhhhhhhhh. ok. does this work with numpy arrays as well? I've only used Sequential models till now ( Even that I just learnt about about a month ago )

File: tensorflow/python/keras/layers/pooling.py [hunk 5]
```diff
@@ -480,7 +480,7 @@ class AveragePooling2D(Pooling2D):
     - If `data_format='channels_first'`:
       4D tensor with shape `(batch_size, channels, pooled_rows, pooled_cols)`.
   
-  >>>Average_pooling_layer = tf.keras.layers.AveragePooling2D(pool_size = (2,2),strides = (1,1),padding = "same")
+  >>>average_pooling_layer = tf.keras.layers.AveragePooling2D(pool_size = (2,2),strides = (1,1),padding = "same")
```
Comment (rpalakkal) [inline tensorflow/python/keras/layers/pooling.py]: Thanks @generationXcode for fixing some of style issues. Not sure if this is necessary, but I think there may need to be a space after >>> . All of @WilliamHYZhang 's requested changes also lgtm. Cheers and keep it up!

Comment (generationXcode) [reply tensorflow/python/keras/layers/pooling.py]: sure

File: tensorflow/python/keras/layers/pooling.py (hunk 4 above)
Comment (generationXcode) [reply tensorflow/python/keras/layers/pooling.py]: Hi I tried what you said in colab and it return `<tf.Tensor 'average_pooling2d_2/Identity:0' shape=(None, 1, 1, 1) dtype=float32>`

Comment (WilliamHYZhang) [reply tensorflow/python/keras/layers/pooling.py]: That's odd, the shapes don't even match up. Maybe I'm understanding it wrong, and in the meantime wait for a reviewer with write access looks at this?

Comment (rpalakkal) [reply tensorflow/python/keras/layers/pooling.py]: I don't think `tf.keras.layers.AveragePooling2D` works directly with numpy arrays since it is a layer of a `tf.keras` model, so an input layer is required (I think the docs say that it takes a 4D tensor as an input)

File: tensorflow/python/keras/layers/pooling.py (hunk 5 above)
Comment (generationXcode) [reply tensorflow/python/keras/layers/pooling.py]: Dit what you said😊

File: tensorflow/python/keras/layers/pooling.py (hunk 4 above)
Comment (generationXcode) [reply tensorflow/python/keras/layers/pooling.py]: I think its to do with the strides and the pool size. I've encountered this before.

Comment (rpalakkal) [reply tensorflow/python/keras/layers/pooling.py]: @WilliamHYZhang I tried it too and got what you got - not sure what's causing the discrepancy

Comment (generationXcode) [reply tensorflow/python/keras/layers/pooling.py]: > I don't think `tf.keras.layers.AveragePooling2D` works directly with numpy arrays since it is a layer of a `tf.keras` model, so an input layer is required (I think the docs say that it takes a 4D tensor as an input)

ohh. ok

File: tensorflow/python/keras/layers/pooling.py (hunk 5 above)
Comment (rpalakkal) [reply tensorflow/python/keras/layers/pooling.py]: Nice job 😃

File: tensorflow/python/keras/layers/pooling.py [hunk 6]
```diff
@@ -446,7 +446,12 @@ def __init__(self,
 @keras_export('keras.layers.AveragePooling2D', 'keras.layers.AvgPool2D')
 class AveragePooling2D(Pooling2D):
   """Average pooling operation for spatial data.
-
+  
+  >>> import numpy as np
```
Comment (rpalakkal) [inline tensorflow/python/keras/layers/pooling.py]: no need to import numpy if you are not using it

File: tensorflow/python/keras/layers/pooling.py (hunk 4 above)
Comment (generationXcode) [reply tensorflow/python/keras/layers/pooling.py]: I tried it again at https://colab.research.google.com/drive/1X2YlL7CK97JibbwinRh7wXMuxhjJp7mu with different strides and pool sizes and got a goo output. Haha theres an advantage in making so many errors in code. I just have a sense now of whats wrong lol

File: tensorflow/python/keras/layers/pooling.py (hunk 6 above)
Comment (generationXcode) [reply tensorflow/python/keras/layers/pooling.py]: ok. I forgot about that. I'll remove it

**Review by rpalakkal — COMMENTED** [review]
LGTM 👍

File: tensorflow/python/keras/layers/pooling.py (hunk 4 above)
Comment (rpalakkal) [reply tensorflow/python/keras/layers/pooling.py]: haha trial and error is my favorite, nice job!

File: tensorflow/python/keras/layers/pooling.py [hunk 7]
```diff
@@ -446,7 +446,11 @@ def __init__(self,
 @keras_export('keras.layers.AveragePooling2D', 'keras.layers.AvgPool2D')
 class AveragePooling2D(Pooling2D):
   """Average pooling operation for spatial data.
-
+  
+  >>> x = tf.keras.Input(shape=(3, 3, 1), name='x')
+  >>> average_pooling_layer = tf.keras.layers.AveragePooling2D(pool_size = (1,1),strides = (1,1),padding = "same")(x)
+  <tf.Tensor 'average_pooling2d_3/Identity:0' shape=(None, 3, 3, 1) dtype=float32>
```
Comment (rthadur) [inline tensorflow/python/keras/layers/pooling.py]: Thanks for your contribution , can you please remove chevron(>>>) symbol  and limit the line length to 80 characters.
Here is the link for [contribution guidelines.](https://github.com/google/styleguide/blob/gh-pages/pyguide.md#google-python-style-guide)

**Review by rthadur — CHANGES_REQUESTED** [review]

**Comment by gbaned** [comment]
@generationzcode Can you please check rthadur's comments and keep us posted? Thanks!

**Comment by tensorflowbutler** [comment]
It has been 15 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by gbaned** [comment]
I'm going to go ahead and close this PR, because it seems to have stalled. If you're still interested in pursing this (and responding to my comments), please feel free to reopen!
